### PR TITLE
websocket support

### LINKED ISSUE #64: push support
Clients poll. A lot. Tornado can do WS oob. Already requires Redis (as a sync backend).

- Dual-homing again, with FCM support as well?
  - In this case it would make sense to spin the FCM backend used in q-drop off in a reusable fashion. Possibly as a separate service?
  - We already saw in q-drop that the FCM HTTP API is slow as buggers. Using the XMPP API (our end is a XMPP client to FCM) may help
  - Along these lines: using custom FCM wrapper based on async HTTP client tech may mitigate the large latency. If XMPP doesn't have that in the first place, then an async XMPP client would likely work even better.
- What needs to be pushed - everything, or is just the DM enough to have the client figure out what has to be pulled?

### PR DESCRIPTION
**Pull request by enkore**
Fixes #64

### CODE REVIEW

File: src/blockserver/tests/test_server.py [hunk 1]
```diff
@@ -368,3 +370,112 @@ def test_database_finish_called_in_quota(backend, http_client, headers, base_url
...
+    assert file_path.startswith('/')
+    assert msg['path'] == file_path[1:]  # file_path is /..., path is just the path, no leading slash
+    assert msg['etag'] == response.headers['ETag']
+
+
+@pytest.mark.gen_test
+def test_ws_prefix_no_authorization(backend, http_server, websocket_prefix_connector):
+    try:
+        yield websocket_prefix_connector()
+        assert False, 'Did not raise'
```
Comment (audax) [inline src/blockserver/tests/test_server.py]: I think this should be in the else part.

Also, is there a way to use pytest.raises instead?

File: src/blockserver/server.py [hunk 2]
```diff
@@ -403,8 +488,22 @@ def get_transfer_cls():
     if database_pool is None:
         database_pool = SimpleConnectionPool(1, 20, dsn=options.psql_dsn)
 
+    prefix = r'(?P<prefix>[\d\w-]+)/'
+    file = r'(?P<file_path>[/\d\w-]+)'
+
     application = Application([
-        (r'^/api/v0/files/(?P<prefix>[\d\w-]+)/(?P<file_path>[/\d\w-]+)', FileHandler, dict(
+        (r'^/api/v0/files/' + prefix + file + '/ws', FileWebSocketHandler, dict(
```
Comment (audax) [inline src/blockserver/server.py]: could this regex result in nasty backtracking performance issues?

Comment (enkore) [inline src/blockserver/server.py]: The regex didn't change. Imho this should be a forward-evaluated re.

File: src/blockserver/tests/test_server.py (hunk 1 above)
Comment (enkore) [inline src/blockserver/tests/test_server.py]: raises doesn't give access to the exception value, but I want to make sure that it doesn't just throw an error, but this exact one.

File: src/blockserver/server.py (hunk 2 above)
Comment (audax) [inline src/blockserver/server.py]: instead of the /ws suffix, use a /ws prefix instead of /files

**Pull request merged**